bugsnag-plugin-react-native/src/main/java/com/bugsnag/android/ThreadDeserializer.java

```diff
@@ -0,0 +1,37 @@
+package com.bugsnag.android;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+class ThreadDeserializer implements MapDeserializer<Thread> {
+
+    private final StackframeDeserializer stackframeDeserializer;
+    private final Logger logger;
+
+    ThreadDeserializer(StackframeDeserializer stackframeDeserializer, Logger logger) {
+        this.stackframeDeserializer = stackframeDeserializer;
+        this.logger = logger;
+    }
+
+    @Override
+    public Thread deserialize(Map<String, Object> map) {
+        String type = MapUtils.getOrThrow(map, "type");
+        List<Map<String, Object>> stacktrace = MapUtils.getOrThrow(map, "stacktrace");
+        List<Stackframe> frames = new ArrayList<>();
+
+        for (Map<String, Object> frame : stacktrace) {
+            frames.add(stackframeDeserializer.deserialize(frame));
+        }
+
+        return new Thread(
```

Comment:
Are we using public or internal version? This is different in `ErrorDeserializer`, so probably needs to be consistent.

Reply:
We are consistently returning the public interface from all the deserializers.

Reply:
Ah yes - OK. I misread Error.